### PR TITLE
En_Gk

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Gk/z_en_gk.c [hunk 1]
```diff
@@ -74,89 +102,1187 @@ static DamageTable D_80B53248 = {
...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gk/func_80B50854.s")
+void EnGk_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnGk* this = THIS;
+    Vec3f sp58 = D_80B533A8;
+    Vec3f sp4C = D_80B533B4;
+    Vec3f sp40 = D_80B533C0;
+    Vec3f sp34 = D_80B533CC;
+    Vec3f sp28 = D_80B533D8;
```
[the remark this answers is not in the document]
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Gk/z_en_gk.c:1119]: They could be, but I was told we need as much in the map as possible.

[the remark this answers is not in the document]
Comment (hensldm) [reply src/overlays/actors/ovl_En_Gk/z_en_gk.c:1120]: I feel we should make an exception for instances like this.

File: src/overlays/actors/ovl_En_Gk/z_en_gk.c [hunk 2]
```diff
@@ -74,89 +102,1184 @@ static DamageTable D_80B53248 = {
...
+
+        Matrix_RotateY(this->actor.shape.rot.y, MTXMODE_NEW);
+        Matrix_MultiplyVector3fByState(&sp4C, &this->unk_30C);
+        Matrix_MultiplyVector3fByState(&sp40, &this->unk_300);
+
+        Matrix_StatePop();
+    }
+}
+
+void EnGk_TransformDraw(GlobalContext* globalCtx, s32 limbIndex, Actor* thisx) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Gk/z_en_gk.c:1163]:
```suggestion
void EnGk_TransformLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Actor* thisx) {
```

File: tools/disasm/functions.txt [hunk 3]
```diff
@@ -14003,9 +14003,9 @@
     0x80B526FC:("EnGk_Init",),
     0x80B529B8:("EnGk_Destroy",),
     0x80B529E4:("EnGk_Update",),
-    0x80B52AD4:("func_80B52AD4",),
-    0x80B52AF0:("func_80B52AF0",),
-    0x80B52D8C:("func_80B52D8C",),
+    0x80B52AD4:("EnGk_OverrideLimbDraw",),
+    0x80B52AF0:("EnGk_PostLimbDraw",),
+    0x80B52D8C:("EnGk_TransformDraw",),
```
Comment (hensldm) [inline tools/disasm/functions.txt:14008]:
```suggestion
    0x80B52D8C:("EnGk_TransformLimbDraw",),
```

File: src/overlays/actors/ovl_En_Gk/z_en_gk.c [hunk 4]
```diff
@@ -74,89 +102,1184 @@ static DamageTable D_80B53248 = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gk/EnGk_Update.s")
+        func_800BC620(&sp5C, &sp50, 255, globalCtx);
+    } else {
+        gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(D_80B533E4[this->unk_2E0]));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gk/func_80B52AD4.s")
+        SkelAnime_DrawTransformFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable,
+                                       this->skelAnime.dListCount, EnGk_OverrideLimbDraw, EnGk_PostLimbDraw,
+                                       EnGk_TransformDraw, &this->actor);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Gk/z_en_gk.c:1261]:
```suggestion
                                       EnGk_TransformLimbDraw, &this->actor);
```

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**